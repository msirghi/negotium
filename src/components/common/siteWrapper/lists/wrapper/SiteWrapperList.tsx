import { FC } from 'react';
import { Typography } from '@mui/material';
import colors from '../../../../../common/styles/colors';

type Props = {
  title?: string;
};

export const SiteWrapperList: FC<Props> = ({ children, title }) => {
  return (
    <div>
      {title && (
        <Typography
          data-testid="site-wrapper-title"
          sx={{
            marginLeft: 10,
            marginTop: 4,
            fontSize: 11,
            textTransform: 'uppercase',
          }}
          color={colors.greys['500']}
        >
          {title}
        </Typography>
      )}
      {children}
    </div>
  );
};
