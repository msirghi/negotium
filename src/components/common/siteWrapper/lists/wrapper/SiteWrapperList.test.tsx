import renderer from 'react-test-renderer';
import { SiteWrapperList } from './SiteWrapperList';
import { render, waitFor } from '@testing-library/react';
import { screen } from '@testing-library/dom';

describe('SiteWrapperList', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(
      <SiteWrapperList>
        <div>test</div>
      </SiteWrapperList>
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render title if provided', async () => {
    render(
      <SiteWrapperList title={'Title'}>
        <div>test</div>
      </SiteWrapperList>
    );

    await waitFor(() => {
      expect(screen.getByText('Title')).toBeInTheDocument();
    });
  });
});
