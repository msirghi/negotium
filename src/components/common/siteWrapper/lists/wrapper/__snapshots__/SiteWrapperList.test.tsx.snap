// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteWrapperList should match the snapshot 1`] = `
<div>
  <div
    className="makeStyles-titleContainer-1"
    onClick={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "flex-start",
        "width": "initial",
      }
    }
  />
  <div>
    test
  </div>
</div>
`;
